text/0000-volta-updates.md

```diff
@@ -0,0 +1,114 @@
+- Managed users: For Enterprise environments, where Volta is installed by IT on users' machines, available to all users of the computer (Each user will still have their own data directory to manage their own tools)
+
+For Individual users, the goal of this feature is to not require any education. The updates will be checked and made available through notifications or automatically applied, requiring nothing new from the user.
+
+For Managed users, there will be a new command: `volta init`. This command will make sure that a user's environment is set up to run Volta and the shims. This command should be idempotent, so that if a user accidentally runs it again, there are no adverse effects. Additionally, we will need to document what files an Enterprise needs to distribute in order to run Volta in a Managed environment.
+
+# Details
+[details]: #details
+
+For Volta updates, there are three distinct sets of changes that need to be considered: Updating the Binaries, Updating the Data Directory, and Updating the Profile.
+
+## Updating the Binaries
+
+The large majority of updates to Volta involve _only_ updating the binaries, without requiring updates to the Data Directory or Profile.
+
+For Managed users, updating the binaries will be handled by the team managing the distribution, making sure that they are always on the latest binaries.
+
+For Individual users, updating the binaries can be handled in 2 phases:
+
+- First, we can provide a notification that the user's version is out-of-date, letting them know that it's time to upgrade. We can also include the platform-specific install instructions in that notification, making it easy for them to act on the notification. This will still require that the user manually update their system, but it will let them know when it is needed and provide guidance on how to affect that update.
+
+- Later, we can implement an automatic updater, which runs in the background and detects when a new version is available, then downloads and unpacks the binaries, replacing the existing ones. This will result in Individual users having an experience that is effectively the same as Managed users: The binaries will be silently updated behind the scenes, without them having to take any action. This should be as atomic of a process as we can make it, so that the user is less likely to notice any strange behaviors because they are in the middle of an update.
+
+For both of these cases, we want to make sure that the checking does not have an adverse effect on the performance of the tools, especially on the performance of shims. For that reason, we likely want to perform the checks and updates in a separate process or thread, so that the main thread can continue to run. This is especially important in offline or low-bandwidth situations, where the check for new versions may take a long time before ultimately failing, and we don't want the user to have to wait.
```

Comment:
We're getting into implementation details, which is probably unnecessary for this RFC, but if you're talking about doing the check in a separate thread, I would just caution that it's important to get the user-visible behavior of the checks right. You don't want even a warning to be interleaved with other user-visible behavior.